lib/db/backend/backend.go

```diff
@@ -157,13 +158,13 @@ type errNotFound struct{}
 func (*errNotFound) Error() string { return "key not found" }
 
 func IsClosed(err error) bool {
-	_, ok := err.(*errClosed)
-	return ok
+	e := &errClosed{}
+	return errors.As(err, &e)
```

Comment:
I think `var e *errClosed` feels more descriptive than the struct literal thing, here it feels like i need to count the ampersands while stating the type explicitly is less thinking

Reply:
Weirdly enought, that change causes test failures but not the panics that are documented on https://golang.org/pkg/errors/#As:

> As panics if target is not a non-nil pointer to either a type that implements error, or to any interface type.

Reply:
So I mean

```
	var e *errClosed
	return errors.As(err, &e)
```

right?

Reply:
Yep, that breaks our unit tests, see below. I don't really get it, as we actually pass a non-nil pointer to a typed nil-pointer implementing error, which to my understanding should fullfill this:

> An error matches target if the error's concrete value is assignable to the value pointed to by target [...]

```
$ go test -race -short ./lib/db/
--- FAIL: TestIgnoredFiles (0.00s)
    db_test.go:40: key not found
--- FAIL: TestUpdate0to3 (0.00s)
    db_test.go:182: key not found
--- FAIL: TestRepairSequence (0.00s)
    db_test.go:327: key not found
--- FAIL: TestDowngrade (0.00s)
    db_test.go:472: key not found
```

Reply:
But that's not possible... https://play.golang.org/p/gy1LtBTh4Sy

Reply:
I agree, the failures happen on exactly this change though. I pushed it so you can see what the change and error are, maybe there's a typo I fail to see? I can't investigate right now, will do so later if the mystery hasn't been lifted until then.